### DISCUSSION TITLE
Identifying client/server @ _app.js

### DISCUSSION THREAD

**Discussion by tjdam** [vercel/next.js#18919] [Help] [answered]
Hi!

So I was playing with msw following the example you have in this repo and I realized that when on _app.js using `typeof window` would work to identify if you are in the server but it never works when you are on the client.

After some digging I found out you can circumvent that by using useEffect, but I don't feel comfortable not understanding why that's the case.

Could anyone explain it to me?

**Comment by jamesmosier** [comment, marked as answer]
Server only: 
```js
// you are on the server because the server has no concept of `window`
if (typeof window === 'undefined') 
```

Client only: 
```js
// you are on the client because the browser has a `window` object
if (typeof window !== 'undefined') 
```

Client only: 
```js
useEffect(() => {
  // only runs on the client 
  // this is the default behavior as defined by React
  // useEffect can only run once the JavaScript is downloaded on the users browser
  // example: https://gist.github.com/gaearon/e7d97cdf38a2907924ea12e4ebdf3c85
}, [])
```
